cmake/checks/check_01_compiler_features.cmake

```diff
@@ -313,3 +313,23 @@ ELSE()
   SET(DEAL_II_DEPRECATED " ")
 ENDIF()
 
+
+#
+# GCC and Clang allow fine grained control of diagnostics via the "GCC
+# diagnostic" pragma. Check whether the compiler supports the "push" and
+# "pop" mechanism and the "ignored" toggle. Further, test for the
+# alternative "_Pragma(...)" variant (and that it does not emit a warning).
+#
+# - Matthias Maier, 2015
+#
+PUSH_CMAKE_REQUIRED("-Werror")
+CHECK_CXX_SOURCE_COMPILES(
+  "
+  _Pragma(\"GCC diagnostic push\")
+  _Pragma(\"GCC diagnostic ignored \\\\\\\"-Wextra\\\\\\\"\")
```

Comment:
@bangerth: The escaping "logic" in cmake is arcane black magic. You would guess that it is correct to match a <code>\"</code> with <code>\\\"</code> - no, no, <code>\\\\\\\"</code> it is. For whatever reason.